dolphinscheduler-task-plugin/dolphinscheduler-task-mlflow/src/main/resources/docker-compose.yml

```diff
@@ -0,0 +1,28 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+version: "3"
+
+services:
+  mlflow-model:
+    image: "${imageName}"
+    ports:
+      - "${deployPort}:8080"
+    deploy:
+      resources:
+        limits:
+          cpus: "${cpuLimit}"
+          memory: "${memoryLimit}"
```

Comment:
I think we should add healthcheck here and use some methods such as ```https://www.testcontainers.org/``` that can be monitored. So that we can ensure the health of the container before performing subsequent operations.

Reply:
the bad news is currently we do not have a monitor for the MLOPS task

Reply:
I think adding another MLOps tool for Monitor is the better way.

Reply:
WDYT @SbloodyS

Reply:
> I think adding another MLOps tool for Monitor is the better way.

It's better if we can monitor it.

Reply:
> > I think adding another MLOps tool for Monitor is the better way.
> 
> It's better if we can monitor it.

I have no idea to add a monitor in the MLflow task plugin because I still don't think the MLflow task plugin should include monitoring docker health.

Reply:
But I think it's reasonable to add more parameters if you need to start the Docker container

Reply:
> I think we should add healthcheck here and use some methods such as `https://www.testcontainers.org/` that can be monitored. So that we can ensure the health of the container before performing subsequent operations.

Currently ```E2E``` using this package to start ```docker container``` from ```docker-compose.yaml``` and monitor it and destory it after all is done.

From the perspective of users using DS. Users do not pay attention to the task type. What they expect is that all task types have the same functionality, including stopping tasks. If the start, stop and running status of containers are not monitored, it is likely that abnormal debug caused by user environment problems will be very difficult. At the same time, it may also cause many orphan containers to affect the user's docker environment.

Reply:
I think your point is reasonable in most tasks. But in the model deployment task, I think state means something like this:
Success: Successfully publish the model.
Running: Deploying the model. In this case, In this case, it means creating the Docker and running the Docker container
Failure: Deployment failure. In this case, maybe an error in the command from MLflow docker build or an error in run $docker compose$ command

So, If the model deployment is finished, we don't need to track the state anymore because it's done @SbloodyS @zhongjiajie

Reply:
I'm not familiar about ML. I'm just asking the usual questions questions.

The command ```docker run``` sometimes would not throw exception if the docker container successfully start but healthcheck failed. After several runs it may cause many orphan containers in the user's host machine and it can't be known by ds or removed by ds. The users can only view the corresponding tasks through ```SSH``` on the worker's machine. Does this lose some features of DS? @jieguangzhou

Reply:
Thank you for your advice. I've added a health check inside the docker container and also added a minute healthy outside the container until the container status becomes healthy. If not healthy, the state in DS will be set as Failure.
@zhongjiajie @SbloodyS